docs/source/topics/events.rst

```diff
@@ -66,3 +68,60 @@ event rule.  For example:
 In the app above, chalice will create two lambda functions,
 and configure ``every_hour`` to be invoked once an hour,
 and ``every_two_hours`` to be invoked once every two hours.
+
+
+.. _s3-events:
+
+S3 Events
+=========
+
+You can configure a lambda function to be invoked whenever
+certain events happen in an S3 bucket.  This uses the
+`event notifications`_ feature provided by Amazon S3.
+
+To configure this, you just tell Chalice the name of an existing
+S3 bucket, along with what events should trigger the lambda function.
+This is done with the :meth:`Chalice.on_s3_event` decorator.
+
+Here's an example:
+
+.. code-block:: python
+
+    import boto3
+    from chalice import Chalice
+
+    app = chalice.Chalice(app_name='s3eventdemo')
+    s3 = boto3.client('s3')
+
+    @app.on_s3_event(bucket='mybucket-name',
+                     events=['s3:ObjectCreated:*'])
+    def handle_s3_event(event):
+        # Download the newly created file locally.
+        s3.downlad_file(event.bucket, event.key, '/tmp/tempfile')
```

Comment:
Typo: `s3.downlad_file` --> `s3.download_file`. It may be good to try both the examples in the reference and topic guide work if you have not already tried them.